### ISSUE TITLE
Add more fields (optional) to the modal

### ISSUE THREAD

**Issue by andergmartins** [OSTraining/OSDownloads#129]
Allow to add custom fields to the modal, not only e-mail.

**Comment by stevejburge** [comment]
Here's a very good use for this feature. From a user ... 

"
For inbound marketing we create resources (infographics, whitepapres, reports, checklists, etc). Visitors (currently-without joomla) need to fill out a form to download the resources. We want to start hosting these resources in our new joomla site but need the following:

1- resources to be fully protected (not accessible to the public)
2- users to fill out a form:
a) either have an autogenerated password sent to their email (make it easier for them), or
b) have the option to register or not. 
In other words, we don´t really need them to create a user & password if they don´t want to, just to fill out the form. They can of course create a user if they wish to (and are re-curring users). "
